### PR TITLE
Add solution for RSpec Calculator

### PR DESCRIPTION
**Pull request by AnnAllan**

### CODE REVIEW

File: spec/calculator_spec.rb [hunk 1]
```diff
@@ -1 +1,154 @@
...
+            expect(c.divide(2, 3)).to be_within(0.5).of(0.666)
+        end
+        it 'raises an ArgumentError if the second argument is zero' do
+            expect{c.divide(2, 0)}.to raise_error(ArgumentError)
+        end
+        it 'returns an integer if there is no remainder' do
+            expect(c.divide(4, 2)).to eq(2)
+        end
+        it 'returns a float if there is a remainder' do
+            expect(c.divide(2, 3)).to be_within(0.5).of(0.666)
```
Comment (remis1889) [inline spec/calculator_spec.rb:59]: you can also check if `2/3` is a `Float`

File: spec/calculator_spec.rb [hunk 2]
```diff
@@ -1 +1,154 @@
...
+        end
+        it 'returns the power of its negative base' do
+            expect(c.pow(-3, 3)).to eq(-27)
+        end
+        it 'returns the power of its negative exponent' do
+            expect(c.pow(2, -2)).to eq(0.25)
+        end
+        it 'returns the power of its decimal arguments' do
+            expect(c.pow(27, 1/3.0)).to eq(3.0)
+        end
```
Comment (remis1889) [inline spec/calculator_spec.rb:76]: also, test the case when exponent is zero

File: spec/calculator_spec.rb [hunk 3]
```diff
@@ -1 +1,154 @@
...
+            expect(c.sqrt(9)).to eq(3)
+        end
+        it 'raises errors for negative arguments' do
+            expect{c.sqrt(-9)}.to raise_error
+        end
+        it 'returns integers for round roots' do
+            expect(c.sqrt(9)).to eq(3)
+        end
+        it 'returns 2-digit decimals for non-round roots' do
+            expect(c.sqrt(8)).to eq(2.83)
```
Comment (remis1889) [inline spec/calculator_spec.rb:93]: check if the result has 2 decimal places exactly

File: spec/calculator_spec.rb (hunk 1 above)
Comment (AnnAllan) [reply spec/calculator_spec.rb:59]: Thank you for all of these reviews.  I will work my way through them to improve my projects.  Completed.

File: spec/calculator_spec.rb (hunk 2 above)
Comment (AnnAllan) [reply spec/calculator_spec.rb:76]: Completed.

File: spec/calculator_spec.rb (hunk 3 above)
Comment (AnnAllan) [reply spec/calculator_spec.rb:93]: Completed.
